pkg/arvo/sur/chat-store.hoon

```diff
@@ -30,16 +30,24 @@
 ::
 +$  chat-configs  (map path config)
 ::
-+$  chat-action
++$  chat-base
   $%  [%create =ship =path]         ::  %create: create a mailbox at ~ship/path
       [%delete =path]               ::  %delete: delete a mailbox at path
       [%message =path =envelope]    ::  %message: append a message to mailbox
       [%read =path]                 ::  %read: set mailbox to read
   ==
 ::
++$  chat-action
+  $%  ::  %messages: append a list of messages to mailbox
+      ::
+      [%messages =path envelopes=(list envelope)]
```

Comment:
Shouldn't `%message` just be deprecated in favor of this?

Reply:
Also updating types that go over the air has historically given problems when ships on latest and latest-1 tried to communicate. Did you test this case on a fakenet, or are these changes going to be breached in?

Reply:
I expect this to work just fine once all ships have updated given that my state import functions have been tested on ships with older state versions. As for this network message issue, seems like it'll only be a problem for ships on latest vs latest-1, which shouldn't last very long?

Reply:
As for %message, I think it's semantically useful to have a "singular version". Idk, arguable.

Reply:
Yeah I don't doubt that app state will be fine.

> it'll only be a problem for ships on latest vs latest-1, which shouldn't last very long?

You think too highly of the current network. (^:

iirc the consequences were pretty harsh last time we messed this up. Big stack traces, unusable chats, etc. Worth testing and maybe even holding off until breach for.

---

> I think it's semantically useful to have a "singular version". Idk, arguable.

If it's semantics, then that difference should be explicit (instead of `%singular` and `%plural`). Implementation-wise, I've found it nice to stick to singular when you can, but just go all-in on plural when you can't. Having both is just interface bloat.

Reply:
This kind of change is supposed to work OTA -- if the message types become incompatible, the subscription is cleared (or the sender backs off, as appropriate). In practice, that process is incomplete and not particularly tuned, so it can be quite noisy.

A simple solution is to just use a different mark for backlog subscriptions. The receiver won't make the subscription if they don't have the mark, and the new type won't mess with current subscriptions. Then you can combine the types in the next breach.

Reply:
This simple solution feels very messy to me... I'd much prefer noise in the dojo than making another mark.

Reply:
Tfw a developer is unwilling to make sacrifices for the experience of their users...

Reply:
Seconding @joemfb's sentiment here. Whatever's messy about it we can clean up on next breach (that is, right now in a separate, breach-flagged PR). We're busy enough with support as-is, don't need more loud errors scaring people.

Reply:
If you can't add a type to a tagged union OTA, then our support for OTAs doesn't exist imo.

Reply:
Call it what you will. I've seen the fall out of not doing this and it sucks and will mean emails that someone else has to respond to. Please do this and let's wrap this up.

Reply:
Working on this next.

Reply:
I've tested it with the new "chat-two" marks, and it all works great.